### ISSUE TITLE
Types of BatchNormalization link are described inaccurately

### ISSUE THREAD

**Issue by gwtnb** [chainer/chainer#2900]
Although `avg_mean` and `avg_var` are `ndarray`, the docstring says that they are `Variable`.

https://github.com/chainer/chainer/blob/6a526e53aee773b65dca7f0383890fa0ad9a9fac/chainer/links/normalization/batch_normalization.py#L56-L57

**Comment by niboshi** [comment]
`avg_mean` and `avg_var` are initialized as `numpy.ndarray`[1] but
`running_avg` and `running_var`[2] could be `cupy.ndarray`.
It it safe?

[1] https://github.com/chainer/chainer/blob/6a526e53aee773b65dca7f0383890fa0ad9a9fac/chainer/links/normalization/batch_normalization.py#L69
[2] https://github.com/chainer/chainer/blob/6a526e53aee773b65dca7f0383890fa0ad9a9fac/chainer/links/normalization/batch_normalization.py#L145
